server/service/users.go

```diff
@@ -59,6 +60,26 @@ func (svc *Service) CreateUser(ctx context.Context, p fleet.UserPayload) (*fleet
 		return nil, ctxerr.Wrap(ctx, err, "verify user payload")
 	}
 
+	if teams != nil {
+		// Validate that the teams exist
+		teamsSummary, err := svc.ds.TeamsSummary(ctx)
+		if err != nil {
+			return nil, ctxerr.Wrap(ctx, err, "fetching teams")
```

Comment:
Should we have a more clear error here?

```suggestion
			return nil, ctxerr.Wrap(ctx, err, "fetching teams in attempt to verify team exists")
```